### ISSUE TITLE
Upload progress jumps to 50% and is always 1 of X

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#1040]
**Describe the bug**
Upload progress doesn't seem to show the real progress. For one thing it immediately jumps to 50%. For another, it never moves from "1 of 5" to "2 of 5" like you'd expect, but seems to always display "1 of 5" then "1 of 4"

**To Reproduce**
1. Import 3 photos as 3 separate observations
2. Tap on "Upload 3 observations" in MyObs
3. Note the following:
    - Linear progress at the bottom of the toolbar immediately goes to 50% even though it says "Uploading 1 of 3 observations"
    - Progress goes from "Uploading 1 of 3 observations" to "Uploading 1 of 2 observations"
    - When upload seems to complete (linear progress bar is at 100%), the status says "Uploading 3 of 0 observations" and the activity indicator is still spinning (goes back when you navigate away and come back)

**Expected behavior**
- [ ] #1167
- [x] Upload progress should keep track of the total number of observations being uploaded so the status should go from "Uploading 1 of 3" to "Uploading 2 of 3" etc
- [x] Upload display should show that upload was complete instead of looking like it goes forever

**Context (please complete the following information):**
 - Device: Pixel 8
 - OS: Android 14
 - App version: [v0.20.0+68](https://github.com/inaturalist/iNaturalistReactNative/releases/tag/v0.20.0%2B68)

**Additional context**
The first two problems happen in iOS 17 too (tried on iPad Air), but the last one seems restricted to Android.

**Comment by kueda** [comment]
Since this is probably related, sometimes the above repro steps start counting up after the total gets to zero, e.g. "1 of 0", "2 of 0", etc. Here's a screen cap Carrie recorded (iPhone 13 Pro, iOS 17.2.1):

Uploading RPReplay_Final1705624839.mov…

**Comment by jtklein** [comment]
For the remaining issue of the progress bar jumping to 50% there are at least two underlying issues here:
1.) In `uploadObservation.js` the `uploadEvidence` function is called twice for the array `unsyncedPhotos`, this leads to four increments of progress being emitted, in contrast to two expected in the `MyObservationsContainer.js`.
2.) Uploads are started in parallel, so the first batch of progress increments (emitted before starting an upload) arrive in parallel for all observations thereby masking the linear progress of finished observations.

**Comment by jtklein** [comment]
Point two is not an issue according to this conversation here: https://inaturalist.slack.com/archives/C02G5AATVGA/p1708712116271609

So, I will close now and turn point one into a separate issue.
